src/antares/model/study.py

```diff
@@ -264,9 +264,14 @@ def create_link(
         area_to: str,
         properties: Optional[LinkProperties] = None,
         ui: Optional[LinkUi] = None,
-        existing_areas: Optional[MappingProxyType[str, Area]] = None,
     ) -> Link:
-        link = self._link_service.create_link(area_from, area_to, properties, ui, existing_areas)
+        missing_areas = [
+            area for area in [area_from, area_to] if area not in self._areas or not hasattr(self._areas[area], 'id')
+        ]
+        if missing_areas:
```

Comment:
We should perform 2 checks:
- The one you did
- Asserts that there's not the same link already created in the study.

Also, you can remove the check with the `hasattr` as it doesn't really make sense IMO